spec.html

```diff
@@ -24176,16 +24390,392 @@ <h1>Runtime Semantics: Evaluation</h1>
     </emu-clause>
   </emu-clause>
 
+  <emu-clause id="sec-decorators">
+    <h1>Decorators</h1>
+    <h2>Syntax</h2>
+    <emu-grammar type="definition">
+      DecoratorList[Yield, Await] :
+        DecoratorList[?Yield, ?Await]? Decorator[?Yield, ?Await]
+
+      Decorator[Yield, Await] :
+        `@` DecoratorMemberExpression[?Yield, ?Await]
+        `@` DecoratorParenthesizedExpression[?Yield, ?Await]
+        `@` DecoratorCallExpression[?Yield, ?Await]
+
+      DecoratorMemberExpression[Yield, Await] :
+        IdentifierReference[?Yield, ?Await]
+        DecoratorMemberExpression[?Yield, ?Await] `.` IdentifierName
+
+      DecoratorParenthesizedExpression[Yield, Await] :
+        `(` Expression[+In, ?Yield, ?Await] `)`
+
+      DecoratorCallExpression[Yield, Await] :
+        DecoratorMemberExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
+        DecoratorParenthesizedExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
```

Comment:
```js
@(a[1])(args)
```

Why do we need this syntax rule? Why not just

```js
@(a[1](args))
```

Reply:
I’m matching the syntax which shipped in Babel here for stage 2 decorators, which is the syntax that was agreed upon before I started working on decorators. Personally wouldn’t mind changing it, I will discuss with the larger group.